libp2p/HostCapability.h

```diff
@@ -36,46 +37,28 @@ namespace p2p
 
 class HostCapabilityFace
 {
-    friend class Host;
-    template <class T> friend class HostCapability;
-    friend class Capability;
-    friend class Session;
-
 public:
-    HostCapabilityFace() {}
-    virtual ~HostCapabilityFace() {}
-
-    Host* host() const { return m_host; }
-
-    std::vector<std::pair<std::shared_ptr<SessionFace>, std::shared_ptr<Peer>>> peerSessions() const;
-    std::vector<std::pair<std::shared_ptr<SessionFace>, std::shared_ptr<Peer>>> peerSessions(u256 const& _version) const;
+    virtual ~HostCapabilityFace() = default;
 
     virtual std::string name() const = 0;
     virtual u256 version() const = 0;
-
-protected:
-    CapDesc capDesc() const { return std::make_pair(name(), version()); }
     virtual unsigned messageCount() const = 0;
-    virtual std::shared_ptr<Capability> newPeerCapability(std::shared_ptr<SessionFace> const& _s, unsigned _idOffset, CapDesc const& _cap) = 0;
 
-    virtual void onStarting() {}
-    virtual void onStopping() {}
+    virtual std::shared_ptr<Capability> newPeerCapability(
+        std::shared_ptr<SessionFace> const& _s, unsigned _idOffset, CapDesc const& _cap) = 0;
 
-private:
-    Host* m_host = nullptr;
+    virtual void onStarting() = 0;
+    virtual void onStopping() = 0;
 };
 
 template<class PeerCap>
 class HostCapability: public HostCapabilityFace
 {
 public:
-    HostCapability() {}
-    virtual ~HostCapability() {}
+    explicit HostCapability(p2p::Host* _host) : m_host(_host) { assert(_host); }
```

Comment:
If possible, the `_host` should be passed by reference here.

Reply:
Changed to reference